starter-code/src/App.js

```diff
@@ -1,21 +1,99 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import contacts from './contacts.json'
+
+
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
-      </div>
-    );
+  constructor(props) {
+      super(props)
+      this.state = {
+          listContacts: contacts.slice(0, 5)
+
+        //  [contacts[0], contacts[1], contacts[2], contacts[3], contacts[4]]
+      }
   }
+
+addRandomContact(){
+let newsList =  this.contacts.slice(0, 6);
```

Comment:
Attention: here you are getting the following error : "TypeError: Cannot read property 'contacts' of undefined".
You should access the state as follows: 
```js
let newsList =  this.state.listContacts.slice(0, 6);
```